src/datetime.rs

```diff
@@ -43,25 +106,81 @@ pub struct DatetimeParseError {
 pub const FIELD: &str = "$__toml_private_datetime";
 pub const NAME: &str = "$__toml_private_Datetime";
 
+/// A parsed TOML date value
+///
+/// May be part of a [`Datetime`]. Alone, `Date` corresponds to a [Local Date].
+/// From the TOML v1.0.0 spec:
+///
+/// > If you include only the date portion of an RFC 3339 formatted date-time,
+/// > it will represent that entire day without any relation to an offset or
+/// > timezone.
+/// >
+/// > ```toml
+/// > ld1 = 1979-05-27
+/// > ```
+///
+/// [Local Date]: https://toml.io/en/v1.0.0#local-date
 #[derive(PartialEq, Clone)]
-struct Date {
-    year: u16,
-    month: u8,
-    day: u8,
+pub struct Date {
+    /// Year: four digits
+    pub year: u16,
+    /// Month: 1 to 12
+    pub month: u8,
+    /// Day: 1 to {28, 29, 30, 31} (based on month/year)
+    pub day: u8,
 }
 
+/// A parsed TOML time value
+///
+/// May be part of a [`Datetime`]. Alone, `Time` corresponds to a [Local Time].
+/// From the TOML v1.0.0 spec:
+///
+/// > If you include only the time portion of an RFC 3339 formatted date-time,
+/// > it will represent that time of day without any relation to a specific
+/// > day or any offset or timezone.
+/// >
+/// > ```toml
+/// > lt1 = 07:32:00
+/// > lt2 = 00:32:00.999999
+/// > ```
+/// >
+/// > Millisecond precision is required. Further precision of fractional
+/// > seconds is implementation-specific. If the value contains greater
+/// > precision than the implementation can support, the additional precision
+/// > must be truncated, not rounded.
+///
+/// [Local Time]: https://toml.io/en/v1.0.0#local-time
 #[derive(PartialEq, Clone)]
-struct Time {
-    hour: u8,
-    minute: u8,
-    second: u8,
-    nanosecond: u32,
+pub struct Time {
+    /// Hour: 0 to 23
+    pub hour: u8,
+    /// Minute: 0 to 59
+    pub minute: u8,
+    /// Second: 0 to {58, 59, 60} (based on leap second rules)
+    pub second: u8,
+    /// Nanosecond: 0 to 999_999_999
+    pub nanosecond: u32,
 }
 
+/// A parsed TOML time offset
+///
 #[derive(PartialEq, Clone)]
-enum Offset {
+pub enum Offset {
+    /// > A suffix which, when applied to a time, denotes a UTC offset of 00:00;
+    /// > often spoken "Zulu" from the ICAO phonetic alphabet representation of
+    /// > the letter "Z". --- [RFC 3339 section 2]
+    ///
+    /// [RFC 3339 section 2]: https://datatracker.ietf.org/doc/html/rfc3339#section-2
     Z,
-    Custom { hours: i8, minutes: u8 },
+
+    /// Offset between local time and UTC
+    Custom {
+        /// Hours: -12 to +12
```

Comment:
It's very minor, but I happened to notice that these docs are slightly incorrect: this can be as large as +13 or even +14 https://en.wikipedia.org/wiki/UTC%2B14:00 for real timezones, on some Pacific islands.

![image](https://user-images.githubusercontent.com/1203825/132611322-21c5007a-5a56-4b27-93b2-82cafd42cdfa.png)![image](https://user-images.githubusercontent.com/1203825/132611271-0c012fe5-840b-457e-be6b-0a59f20857d1.png)

A special mention for Chatham Islands during DST, where the offset is +13:45.

Reply:
Filed #444

Reply:
@huonw Nice find! This made my day on GitHub for sure!